test_todo.py

```diff
@@ -48,3 +55,19 @@ def test_cannot_mark_non_existing_task_complete(todo):
     todo.add_todo("read")
     with pytest.raises(KeyError):
         todo.mark_complete(2)
+
+
+def test_increases_count_to_next_available_id(todo):
+    todo.add_todo("read")
+    todo.add_todo("gym")
+    todo.add_todo("sleep")
+    todo.remove_todo(3)
+    todo.add_todo("sleep")
+    assert list(todo.get_todos().keys()) == [1, 2, 3]
```

Comment:
not 1, 2, 4

also when hooking this up with the command line, self.count (previous version) would be reset to 0 when making a new instance, so I think this is more robust